### PR TITLE
Support pushing chassis config via gNMI root path

### PR DESCRIPTION
**Pull request by Yi-Tseng**

### CODE REVIEW

File: stratum/hal/lib/common/yang_parse_tree.cc [hunk 1]
```diff
@@ -146,6 +146,18 @@ void YangParseTree::SendNotification(const GnmiEventPtr& event) {
...
+
+  // Add the minimum nodes:
+  //   /interfaces/interface[name=*]/state/ifindex
+  //   /interfaces/interface[name=*]/state/name
+  //   /interfaces/interface/...
+  //   /
+  // The rest of nodes will be added once the config is pushed.
+  AddSubtreeAllInterfaces();
+  AddSubtreeAllComponents();
+  AddRoot();
```
Comment (pudelkoM) [inline stratum/hal/lib/common/yang_parse_tree.cc:160]: Also called later https://github.com/stratum/stratum/pull/178/files#diff-c4a583ef9a466f4a5199f2fe94500d9fR229
Do we need both?

Comment (Yi-Tseng) [reply stratum/hal/lib/common/yang_parse_tree.cc:160]: good catch, I think we can put them together, I'll remove the line 229

File: stratum/hal/lib/common/yang_parse_tree_paths.cc [hunk 2]
```diff
@@ -782,12 +783,16 @@ void SetUpRoot(TreeNode* node, YangParseTree* tree) {
...
-
+    // Deserialize the input proto to OpenConfig device format.
+    if (in.ParseFromString(typed_value->bytes_val())) {
+      // Convert the input proto into the internal format.
+      ASSIGN_OR_RETURN(*config->writable(),
+          OpenconfigConverter::OcDeviceToChassisConfig(in));
+    } else {
+      // Try parse it with ChassisConfig format.
+      RETURN_IF_ERROR(
+        ParseProtoFromString(typed_value->bytes_val(), config->writable()));
```
Comment (Yi-Tseng) [inline stratum/hal/lib/common/yang_parse_tree_paths.cc:794]: indent

**Comment by pudelkoM** [comment]
@Yi-Tseng 
Can you give an example how to push a ChassisConfig with the gnmi cli?

File: stratum/hal/lib/common/yang_parse_tree.cc (hunk 1 above)
Comment (pudelkoM) [inline stratum/hal/lib/common/yang_parse_tree.cc:160]: Could this be moved to the end of function? Where `AddRoot()` was before.

**Pull request merged**